### PR TITLE
Adding more config test and tests organization

### PR DESCRIPTION
**Pull request by carlosdlr**

### CODE REVIEW

File: tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/config/retry/CustomRetryServiceException.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2017 Contributors to the Eclipse Foundation
```
Comment (Emily-Jiang) [inline tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/config/retry/CustomRetryServiceException.java]: 2017->2020

File: tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/config/retry/ConfigPropertyGlobalVsClassVsMethodTest.java [hunk 2]
```diff
@@ -40,11 +42,15 @@
...
-                "org.eclipse.microprofile.fault.tolerance.tck.config.BeanWithRetry/Retry/maxRetries=5" +
-                    "\norg.eclipse.microprofile.fault.tolerance.tck.config.BeanWithRetry/triggerException/Retry/maxRetries=6" +
-                    "\nRetry/maxRetries=7"), "microprofile-config.properties")
+            .addClasses(BeanWithRetry.class, TestConfigExceptionA.class, CustomRetryServiceException.class)
+            .addAsManifestResource(
+                new ConfigAnnotationAsset()
+                    .setGlobally(Retry.class, "maxRetries", "5")
+                    .set(BeanWithRetry.class, "triggerException", Retry.class, "maxRetries", "6")
+                    .set(BeanWithRetry.class, "triggerException", Retry.class, "maxRetries", "7")
+                    .set(BeanWithRetry.class, "doProcessWithCustomException", Retry.class, "retryOn",
```
Comment (Emily-Jiang) [inline tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/config/retry/ConfigPropertyGlobalVsClassVsMethodTest.java]: This one is supposed to be a global setting not directly on the class level.

File: tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/config/retry/ConfigPropertyGlobalVsClassVsMethodTest.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017 Contributors to the Eclipse Foundation
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
```
Comment (Emily-Jiang) [inline tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/config/retry/ConfigPropertyGlobalVsClassVsMethodTest.java]: You should not change the starting year of the copyright but just the ending year. It should be 2017-2020.

File: tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/config/retry/ConfigPropertyGlobalVsClassVsMethodTest.java [hunk 4]
```diff
@@ -65,7 +71,21 @@ void propertyPriorityTest() {
...
+    /**
+     * Test that config property on file has precedence over annotation config
+     */
+    @Test
+    void propertyRetryOnPriorityTest() {
+        try {
+            bean.doProcessWithCustomException();
+        }
+        catch (CustomRetryServiceException e) {
+            Assert.assertTrue(e instanceof Exception);
```
Comment (Ladicek) [inline tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/config/retry/ConfigPropertyGlobalVsClassVsMethodTest.java]: I think what you really want here is an empty `catch` block, but a `fail()` call in the `try` block, after `bean.doProcessWithCustomException()`.

**Comment by carlosdlr** [comment]
This was done in the [](https://github.com/eclipse/microprofile-fault-tolerance/pull/500)

**Comment by Ladicek** [comment]
I don't really think so. This PR does a bunch of unnecessary code shuffling, and really adds one test -- for configuring `@Retry.retryOn`.

**Pull request closed without merging**